pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,402 @@
+            The name of the axis to select.
+        range (list, None):
+            The range to set to the axis to selected.
+            If None: autorange will be enabled.
+            If list of len 1: range will be set between 0 and range[0]
+            If list of len 2: range will be set between the two range values
+        """
+        if range is None or len(range) == 0:
+            # AUTORANGE
+            range = None
+        elif len(range) == 1:
+            # ZERO TO R
+            range = [min(0, *range), max(0, *range)]
+        elif len(range) == 2:
+            # SET GIVEN RANGE
+            range = [min(*range), max(*range)]
+        else:
+            raise AttributeError("bad range")
+        if range is None:
+            self.enableAxisAutoRange(axis_name)
+        else:
+            self.disableAxisAutoRange(axis_name)
+            axis = self.axes[axis_name]
+            charts = [self.charts[chart] for chart in axis.charts]
+            if axis.orientation in ["top", "bottom"]:  # IS X AXIS
+                for chart in charts:
+                    vb = chart.plotItem.vb
+                    vb.setXRange(*range, **kwargs)
+            elif axis.orientation in ["left", "right"]:  # IS Y AXIS
+                for chart in charts:
+                    vb = chart.plotItem.vb
+                    vb.setYRange(*range, **kwargs)
+
+    def update(self):
+        """Updates all charts' contents."""
+        for axis_name, axis in self.axes.items():
+            if axis.autorange:
+                charts = [self.charts[chart] for chart in axis.charts]
+                bounds = []
+                if axis.orientation in ["top", "bottom"]:  # IS X AXIS
+                    for chart in charts:
+                        bounds += chart.dataBounds(ViewBox.XAxis)
+                    bounds = [bound for bound in bounds if bound is not None]
+                    if len(bounds) > 0:
+                        for chart in charts:
+                            vb = chart.plotItem.vb
+                            vb.setXRange(min(bounds), max(bounds))
+                elif axis.orientation in ["left", "right"]:  # IS Y AXIS
+                    for chart in charts:
+                        bounds += chart.dataBounds(ViewBox.YAxis)
+                    bounds = [bound for bound in bounds if bound is not None]
+                    if len(bounds) > 0:
+                        for chart in charts:
+                            vb = chart.plotItem.vb
+                            vb.setYRange(min(bounds), max(bounds))
+        super().update()
+
+    def enableAxisAutoRange(self, axis_name):
+        """Enables autorange for the axis with given name.
+
+        Parameters:
+        axis_name (str, None):
+            The name of the axis to select.
+        """
+        self.axes[axis_name].autorange = True
+
+    def disableAxisAutoRange(self, axis_name):
+        """Disables autorange for the axis with given name.
```

Comment:
The following docstring will be warning free:

```python
        """Disables autorange for the axis with given name.

        Parameters
        ----------
        axis_name : str or None
            The name of the axis to select.
        """
```

Reply:
done